src/components/Invoice/use-create-invoice.ts

```diff
@@ -0,0 +1,43 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import type { IInvoiceForm } from '../Models/Form/InvoiceForm.interface';
+import { invoiceFormSchema } from '../Models/Form/InvoiceForm.interface';
+import { useNotifications } from '../NotificationContext/NotificationContext';
+import { useNavigate } from 'react-router-dom';
+
+const createInvoice = (data: IInvoiceForm) =>
+  fetch(`/api/invoices`, {
+    method: 'POST',
+    body: JSON.stringify(data),
+    headers: { 'Content-Type': 'application/json' },
+  })
+    .then((res) => {
+      if (res.status > 399) {
+        throw new Error(`Request failed with status ${res.status}`);
+      }
+
+      return res;
+    })
+    .then((res) => res.json())
+    .then((res) => invoiceFormSchema.parse(res));
+
+export const useCreateInvoice = () => {
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const { setNotification } = useNotifications();
+
+  return useMutation({
+    mutationFn: createInvoice,
+    onSuccess: (response) => {
+      queryClient.invalidateQueries({ queryKey: ['invoices'] });
```

Comment:
invalidateQueries returns a promise, so it might be useful to await it before performing other operations.